en/material-system/effect-syntax.md

```diff
@@ -0,0 +1,259 @@
+# Effect Syntax Guide
+
+Writing your own effect can give you all the capabilities to customize the rendering process.<br>
+Cocos Effect is a single-source embedded domain-specific language, based on YAML and GLSL.<br>
+The YAML part declares the general framework, while GLSL part specifies the actual shader.<br>
+Together they form a complete specification for the rendering process.
+
+We recommend editing effect files using Visual Studio Code with the official `Cocos Effect` plugin from marketplace.
+
+Note: This document is targeted at Technical Artists or Graphics Developers,<br>
+if you are design artists who need specific shader customization needs, please contact your TA or programmer for support and give them this doc.
```

Comment:
Nope, should be **graphics developer**. This is a crucial misunderstanding, since the whole document is targeted at people at least familiar with the basic concepts of computer graphics. (pipeline states, basic shader arithmetics, etc.) Technical artists and graphics developer are the usual job titles for this kind of developer.

Reply:
We are on the 1.1 branch now. I will submit this change to PR https://github.com/cocos-creator/docs-3d/pull/302